File: smoke_tests/csharp/command_injection.cs
// Command Injection vulnerabilities in C#
using System;
using System.Diagnostics;

public class CommandInjectionVulnerabilities
{
    public void ExecuteCommandUnsafe(string userInput)
    {
        // VULNERABLE: Command injection via Process.Start
        Process.Start("cmd.exe", "/c " + userInput);
    }

    public string RunPingUnsafe(string host)
    {
        // VULNERABLE: Command injection in ping
        ProcessStartInfo psi = new ProcessStartInfo
        {
            FileName = "cmd.exe",
            Arguments = "/c ping " + host,
            RedirectStandardOutput = true
        };
        Process p = Process.Start(psi);
        return p.StandardOutput.ReadToEnd();
    }

    public void ExecuteScriptUnsafe(string scriptPath)
    {
        // VULNERABLE: Arbitrary script execution
        Process.Start("powershell.exe", "-ExecutionPolicy Bypass -File " + scriptPath);
    }

    public void ShellExecuteUnsafe(string command)
    {
        // VULNERABLE: Shell execution with user input
        ProcessStartInfo startInfo = new ProcessStartInfo
        {
            FileName = "/bin/bash",
            Arguments = $"-c \"{command}\"",
            UseShellExecute = false
        };
        Process.Start(startInfo);
    }
}

File: smoke_tests/rust/command_injection.rs
// Command Injection vulnerability in Rust
use std::process::Command;

fn run_command_unsafe(user_input: &str) {
    // VULNERABLE: User input passed directly to shell
    Command::new("sh")
        .arg("-c")
        .arg(format!("ls {}", user_input))
        .output()
        .expect("Failed to execute command");
}

fn execute_script_unsafe(script_name: &str) {
    // VULNERABLE: Unsanitized script execution
    Command::new("bash")
        .arg(script_name)
        .spawn()
        .expect("Failed to run script");
}

fn ping_host_unsafe(host: &str) {
    // VULNERABLE: Command injection via ping
    let cmd = format!("ping -c 1 {}", host);
    Command::new("sh")
        .arg("-c")
        .arg(&cmd)
        .output()
        .expect("Ping failed");
}
